packages/paste-core/components/button/src/index.tsx

```diff
@@ -54,25 +59,50 @@ const getButtonState = (disabled?: boolean, loading?: boolean): ButtonStates =>
   return 'default';
 };
 
-const handlePropValidation = ({as, href, tabIndex, variant, size, fullWidth, children}: ButtonProps): void => {
+const handlePropValidation = ({
+  as,
+  href,
+  tabIndex,
+  variant,
+  size,
+  fullWidth,
+  children,
+  disabled,
+  loading,
+}: ButtonProps): void => {
   const hasHref = href != null && href !== '';
   const hasTabIndex = tabIndex != null;
 
+  // Link validation
   if (as !== 'a' && hasHref) {
     throw new Error(`[Paste: Button] You cannot pass href into a button without the 'a' tag.  Use 'as="a"'.`);
   }
-  if (as === 'a' && !hasHref) {
-    throw new Error(`[Paste: Button] Missing href prop for link button.`);
-  }
-  if (as === 'a' && variant === 'link') {
-    throw new Error(`[Paste: Button] This should be a link. Use the [Paste: Anchor] component.`);
+  if (as === 'a') {
+    if (!hasHref) {
+      throw new Error(`[Paste: Button] Missing href prop for link button.`);
+    }
+    if (variant === 'link' || variant === 'inverse_link') {
+      throw new Error(`[Paste: Button] Using Button component as an Anchor. Use the Paste Anchor component instead.`);
+    }
+    if (variant !== 'primary' && variant !== 'secondary' && variant !== 'reset') {
+      throw new Error(`[Paste: Button] <Button as="a"> only works with the following variants: primary or secondary.`);
+    }
+    if (disabled || loading) {
+      throw new Error(`[Paste: Button] <Button as="a"> cannot be disabled or loading.`);
+    }
```

Comment:
We should remember to call this out in the breaking changes as throwing new errors for potentially existing implementations might trip people up who aren't testing thoroughly